packages/plugins/typescript/operations/src/index.ts

```diff
@@ -42,6 +42,20 @@ export interface TypeScriptDocumentsPluginConfig extends RawDocumentsConfig {
    * ```
    */
   immutableTypes?: boolean;
+  /**
+   * @name forceTypename
+   * @type boolean
+   * @default false
+   * @description Automatically adds `__typename` field to the generated types, even when they are not specified
+   * in the selection set, and makes it non-optional
+   *
+   * @example
+   * ```yml
+   * config:
+   *   forceTypename: true
+   * ```
+   */
+  forceTypename?: boolean;
```

Comment:
I think that now the `BaseVisitor` is the one that manages the configuration for `__typename`, this should be defined there (and this config interface inherits from it, so it will be included here as well).

Reply:
See this: https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/other/visitor-plugin-common/src/base-visitor.ts#L87-L100